src/assets.rs

```diff
@@ -195,23 +313,55 @@ impl<'a> AssetEditor<'a> {
     ///   * the given asset has the corresponding field.
     /// * Updates the mint information, if present in the given asset.
     /// * Sets as verified if either asset if verified.
-    pub fn update(&mut self, other: Asset) {
-        assert_eq!(self.asset.definition, other.definition);
-        if other.verified || !self.asset.verified {
+    pub fn update<L: Ledger>(mut self, other: Asset) -> Result<Self, KeystoreError<L>> {
+        let mut asset = self.store.load(&other.definition.code)?;
+        if other.verified || !asset.verified {
```

Comment:
I am getting a bit confused by these `if asset.verified` checks.  Why do we only update some fields if the new Asset is verified of the old one was not verified?  I guess this just excludes the case where other is not verified and the asset in storage is verified?  But in that case why do we still allow updating the mint info?

Reply:
Yeah, we want to avoid overwriting the metadata of a verified asset in the store with unverified info. The mint info is special because a verified asset initially doesn't have a mint info. When creating a verified asset library, the mint info is removed by calling `export_verified`, so we need the mint info from `other`.